### PR TITLE
47 infestation spawn balance

### PR DESCRIPTION
**Pull request by avelitj12**
Raised base infestation rate slightly

Start of game, has a lull until first dome infested (allows initial setup of squad positions)

Once first dome infested, random infestation chance doubles, scaling down based on how many domes are infested. This, in theory allows player to focus on exterminating based on infestation levels and roots, but keeps them on their toes until the cleanse win condition is available (after 3 mins).

After 3 minutes, random infestation chance is removed.

If player has not lost any domes after 3 minutes, they can now systematically exterminate to get a cleanse win (if their resources aren't exhausted!)

Also, minor bugfix for squad resource usage. Pyros and engis were adding resources rather than removing

### CODE REVIEW

**Comment by cmavelis** [comment]
have you tried this? feels unwinnable

File: autoloads/Globals.gd [hunk 1]
```diff
@@ -31,8 +31,8 @@ const RESOUCE_TIMER_DURATION = 1/_RESOURCES_PER_SECOND
 # squad
 var BASE_MOVE_SPEED = 50 
 var BASE_INFESTATION_FIGHT_RATE = -BASE_DOME_INFESTATION_RATE
-var PYRO_SPECIAL_FUEL_USAGE = 3
-var ENGI_SPECIAL_PARTS_USAGE = 3
+var PYRO_SPECIAL_FUEL_USAGE = -2
+var ENGI_SPECIAL_PARTS_USAGE = -2
```
Comment (cmavelis) [inline autoloads/Globals.gd:35]: lol no wonder i never ran out

**Comment by avelitj12** [comment]
> have you tried this? feels unwinnable

I have but wasn't in the position to notice either way, perhaps bringing the infestation chance back down to .008 and extending the cleanse timer to 5 minutes will help

**Comment by avelitj12** [comment]
Adjusted the numbers so cleansing is winnable

**Pull request merged**